src/file/executable_path.cr

```diff
@@ -0,0 +1,106 @@
+# Reference:
+# - https://github.com/gpakosz/whereami/blob/master/src/whereami.c
+# - http://stackoverflow.com/questions/1023306/finding-current-executables-path-without-proc-self-exe
+
+class File
+  PATH_DELIMITER = {% if flag?(:windows) %} ';' {% else %} ':' {% end %}
+
+  # :nodoc:
+  INITIAL_PATH = ENV["PATH"]
```

Comment:
Is `PATH` always present? Maybe, just in case, we can use `ENV["PATH"]?` (same below) and return `nil` if it's not set.